app/models/user.rb

```diff
@@ -7,17 +7,64 @@ class User < ApplicationRecord
   # end
 
   def starred_repos
-    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/starred"))
+    # parse(Faraday.get("https://api.github.com/users/#{self[:username]}/starred"))
+    parse(Faraday.get("https://api.github.com/#{self[:username]}/starred?token=#{self.token}"))
   end
 
   def followers
-   parse(Faraday.get("https://api.github.com/users/#{self[:username]}/followers"))
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+   parse(Faraday.get("https://api.github.com/users/#{self[:username]}/followers#{auth}"))
   end
-  
+
   def following
-  parse(Faraday.get("https://api.github.com/users/#{self[:username]}/following"))
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/following#{auth}"))
+  end
+
+  def repos
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/repos#{auth}"))
+  end
+
+  def organizations
+    # connection = Faraday.new(url: "https://api.github.com") do |faraday|
+    #   faraday.params[:access_token] = self.token
+    # end
+    # response = connection.get "#{User.find_by(token: self.token)}/orgs"
+    # # try = parse(response.body)
+    # byebug
+
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/orgs#{auth}"))
   end
 
+
+  def activity
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    parse(Faraday.get("https://api.github.com/users/#{self[:username]}/events#{auth}"))
+  end
+
+  def activity_of_followed
+    client_id     = ENV["GITHUB_CLIENT_ID"]
+    client_secret = ENV["GITHUB_SECRET_KEY"]
+    auth          = "?client_id=#{client_id}&client_secret=#{client_secret}"
+    # parse(Faraday.get("https://api.github.com/users/#{self[:username]}/received_events#{auth}"))
+    self.following.each do |f|
+      return parse(Faraday.get("https://api.github.com/users/#{f[:login]}/events#{auth}"))
+    end
+  end
```

Comment:
@AliSchlereth received_events returns all of the events connected to either a user or a repo which the current_user follows. However the spec only specifies returning the commits by followed users, not the repo. I wasn't sure if it was ok to also have the activity of repos followed by the user, to I wrote the method a second way that only returns the events of other users. 

@s-espinosa see above for question re: serializing a PORO object before it's received by the view.